### ISSUE TITLE
All messages get marked as read after joining community

### ISSUE THREAD

**Issue by qoqobolo** [status-im/status-react#20712]
# Bug Report

## Problem 

Once the joining status is changed from `Pending` to `Joined`, all messages are automatically marked as read.

https://github.com/status-im/status-mobile/assets/67952253/e431925e-5c1c-4f56-8a24-f91a8028ea31

#### Expected behavior
Messages are not marked as read 

#### Actual behavior
Messages are marked as read 

### Reproduction

1. Open a community with unread messages 
2. Send a request to join
3. Check the unread indicators on the channels after joining

### Additional Information

- Logs: 
[markedasread.zip](https://github.com/user-attachments/files/16164500/markedasread.zip)

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

**Comment by churik** [comment]
rechecked it; yes, all previous messages marked as "read", new one arrive correctly and mark as unread

Postponing to 2.31

**Comment by ulisesmac** [comment]
An update:

It's been really hard to replicate in a development environment, for some reason, when the admin is on a physical device or Status Desktop and the user requesting the access is on an emulator, the response never comes (or maybe takes more than 3 hours).

Additionally,
 **the issue about "marking all community chats as read" is also present when the user opens a community, logs out and then logs in** (this made the testing of the original issue harder). Here's a demo using two emulators (white: regular user | dark: community admin):

https://github.com/user-attachments/assets/1e7da50f-8f80-46b6-a5f3-4e6e1355d607
 
 Regarding to the main issue, we aren't calling any waku RPC to mark the community chats as read, seems this issue comes from status-go, I was also able to reproduce it on Status Desktop. Probably once it's been fixed in desktop it'll be automatically fixed on mobile.
 
 @churik do you know if we have an issue for it on desktop?
 
 CC: @ilmotta What's the best way to address this issue?

**Comment by ilmotta** [comment]
> I was also able to reproduce it on Status Desktop. Probably once it's been fixed in desktop it'll be automatically fixed on mobile.

Nice @ulisesmac 👍🏼 If the problem exists on desktop, it's worth checking with them in their Discord channels to find if the issue is already defined for their client. In theory the issue we need is already created, and could be this one :)

> What's the best way to address this issue?

Probably the fix involves writing a unit test in status-go to reproduce the bug and fix from there. If you are not comfortable with doing that maybe a desktop dev can help pinpoint where the culprit line of code is. Maybe in `#status-go-guild` they can give you a direction in case you want to try to fix this yourself.

In case you don't want to dive into status-go just yet, no worries. I think this bug is relatively harmless and users can live with it for some time.

**Comment by churik** [comment]
@ulisesmac 
yes, bug is reproducible on status-desktop as well, thank you for checking it

**Comment by churik** [comment]
cleaning milestone, as issue does not have high priority

**Comment by ulisesmac** [comment]
Marked as blocked since the requester is stuck on a  "Pending" state to join a community, tried with both mobile and desktop clients and the state is never updated to either rejected or accepted.

Probably related to message reliability, and probably also related to:
- https://github.com/status-im/status-mobile/issues/19575

Can't continue since we first need to sucessfully join to a community. The testing takes a lot of time due to the difficulty of the reproduction

**Comment by ulisesmac** [comment]
For reference, the community manager always has the "accept pending" or "reject pending" state:
![Image](https://github.com/user-attachments/assets/7459d775-f4fa-401b-9c89-c1343cbe6020)

**Comment by churik** [comment]
Can you please try it with a new community (so open fresh community on status-desktop)?
Not the case for me on desktop 2.31 / mobile nightly 26/11.

I don't think that we have a capacity for it in 2.32, given the desired release cut date.
Deferring the issue, please be free to move it bak if you think there is a room for it @ilmotta
